home-app/bundle.js

```diff
@@ -0,0 +1 @@
h>1?{id:r.id,isLocalOnly:!0,isProxy:!0}:{id:r.id||"deviceId",verificationId:r.id}}}};return console.log("IDENTIFY response "+JSON.stringify(n,null,2)),n},this.reachableDevicesHandler=async e=>{console.log("REACHABLE_DEVICES request "+JSON.stringify(e,null,2));const t=e.inputs[0].payload.device.id,r=e.devices.flatMap(e=>{const r=e.customData;return r.proxy===t?[{verificationId:`${t}-${r.channel}`}]:[]}),n={intent:smarthome.Intents.REACHABLE_DEVICES,requestId:e.requestId,payload:{devices:r}};return console.log("REACHABLE_DEVICES response "+JSON.stringify(n,null,2)),n},this.executeHandler=async e=>{console.log("EXECUTE request: "+JSON.stringify(e,null,2));const t=e.inputs[0].payload.commands[0],r=t.execution[0];if("action.devices.commands.ColorAbsolute"!==r.command)throw Error("Unsupported command: "+r.command);const n=(new smarthome.Execute.Response.Builder).setRequestId(e.requestId);return await Promise.all(t.devices.map(async t=>{const o=u(),i=r.params,s=t.customData,a=i.color.spectrumRGB,l=Buffer.alloc(3*s.leds);for(let e=0;e<l.length;e+=3)l.writeUInt8(a>>16&255,e+0),l.writeUInt8(a>>8&255,e+1),l.writeUInt8(a>>0&255,e+2);o.writePixels(s.channel,l);const f=o.read();console.debug("opcMessage:",f);const p=c(s.control_protocol,f);p.requestId=e.requestId,p.deviceId=t.id,p.port=s.port,console.debug(s.control_protocol+" RequestData: ",p);try{const e=await this.app.getDeviceManager().send(p),t={...i,online:!0};n.setSuccessState(e.deviceId,t)}catch(e){n.setErrorState(t.id,e.errorCode)}})),console.log("EXECUTE response: "+JSON.stringify(n,null,2)),n.build()},this.getDiscoveryData=async(e,t)=>{var r,n;if(void 0!==e.udpScanData)return a.a.decodeFirst(Buffer.from(e.udpScanData.data,"hex"));if(void 0!==e.mdnsScanData){const t=e.mdnsScanData;return{id:t.txt.id,model:t.txt.model,hw_rev:t.txt.hw_rev,fw_rev:t.txt.fw_rev,channels:t.txt.channels.split(",").map(e=>parseInt(e,10))}}if(void 0!==e.upnpScanData){const o=e.upnpScanData,s=l(o.location);s.requestId=t,s.deviceId="",s.port=o.port,console.debug("UPnP HTTP command: ",s);try{const e=(await this.app.getDeviceManager().send(s)).httpResponse.body;console.log("XML device description",e);const t=(new i.DOMParser).parseFromString(e,"text/xml").getElementsByTagName("device")[0],o=(null===(n=null===(r=t.getElementsByTagName("UDN")[0].textContent)||void 0===r?void 0:r.match(/uuid:([a-zA-Z0-9]+)/))||void 0===n?void 0:n[1])||"",a=t.getElementsByTagName("modelName")[0].textContent||"",u=t.getElementsByTagName("service"),c=Array.from(u).map(e=>{var t;const r=null===(t=e.getElementsByTagName("serviceId")[0].textContent)||void 0===t?void 0:t.match(/urn:sample:serviceId:strand-([0-9]+)/);return r?parseInt(r[1],10):0});return{id:o,model:a,channels:c}}catch(e){console.log("UPnP HTTP error: ",e)}}throw Error("Missing or incorrect scan data for intent requestId "+t)},this.app=e}}(f);f.onIdentify(p.identifyHandler).onReachableDevices(p.reachableDevicesHandler).onExecute(p.executeHandler).listen().then(()=>{console.log("Ready")})}]);
```

Comment:
Do we need this file? Should it be removed and marked in `.gitignore`?

Reply:
Yeah.  Will remove.